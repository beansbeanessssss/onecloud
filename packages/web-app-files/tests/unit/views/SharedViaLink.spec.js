describe('SharedViaLink view', () => {
  describe('when the page has loaded successfully', () => {
    it.todo('should load the resources')
    it.todo('should adjust the table header position')
  })

  describe('when the view is still loading', () => {
    it.todo('should show list-loader component')
  })

  describe('when the view is not loading anymore', () => {
    it.todo('should not show list-loader component')

    describe('when there are no files to be displayed', () => {
      it.todo('should show no-content-message component')
      it.todo('should not show resource-table component')
    })

    describe('when there are one or more files to be displayed', () => {
      it.todo('should not show no-content-message component')
      it.todo('should show resource-table component with props')
      it.todo('should set props on context-actions component')
      it.todo('should set props on list-info component')
      it.todo('should trigger showing the sidebar when a "showDetails" event gets emitted')
      it.todo('should trigger the default action when a "fileClick" event gets emitted')
      it.todo('should lazily load previews when a "rowMounted" event gets emitted')
    })
  })
})
