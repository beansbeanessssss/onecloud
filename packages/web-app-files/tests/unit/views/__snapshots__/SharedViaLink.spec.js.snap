// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedViaLink view when the view is not loading anymore when there are no files to be displayed should show no-content-message component 1`] = `
<div>
  <div class="uk-height-1-1 uk-flex uk-flex-column uk-flex-center uk-flex-middle uk-text-center files-empty" id="files-shared-via-link-empty">
    <div class="oc-icon oc-icon-xxl oc-icon-passive">
      <!---->
    </div>
    <div class="oc-text-muted uk-text-large"><span data-msgid="There are no resources with a public link at the moment" data-current-language="en_US">There are no resources with a public link at the moment</span></div>
    <div class="oc-text-muted"></div>
  </div>
</div>
`;
